include/swift/Driver/ToolChain.h

```diff
@@ -249,6 +249,14 @@ class ToolChain {
   const Driver &getDriver() const { return D; }
   const llvm::Triple &getTriple() const { return Triple; }
 
+  static void addLinkedLibArgs(const llvm::opt::ArgList &Args,
+                               llvm::opt::ArgStringList &FrontendArgs);
+
+  static void addArgsFromGroupExcept(const llvm::opt::ArgList &Args,
+                                     llvm::opt::ArgStringList &FrontendArgs,
+                                     llvm::opt::OptSpecifier groupID,
+                                     llvm::opt::OptSpecifier exceptID);
```

Comment:
I think you could remove this function in favor of this method on `ArgList`.

```
  /// AddAllArgsExcept - Render all arguments matching any of the given ids
  /// and not matching any of the excluded ids.
  void AddAllArgsExcept(ArgStringList &Output, ArrayRef<OptSpecifier> Ids,
                        ArrayRef<OptSpecifier> ExcludeIds) const;
```

Reply:
Oh ya I tried this already but didn't get the intended output, turns out I just understood ArrayRef wrongly. Will remove this and use that instead.